### PR TITLE
Add ListKeys support

### PR DESCRIPTION
**Pull request by joechrisellis**
Rust client support for the `ListKeys` operation.

### CODE REVIEW

**Comment by joechrisellis** [comment]
NOTE: won't pass CI because this is pending a version bump to `parsec-interface-rs`.

**Review by hug-dev — COMMENTED** [review]
That looks good to me! Could you please just add a test in `src/core/testing/core_tests.rs`?

File: Cargo.toml [hunk 1]
```diff
@@ -26,3 +26,6 @@ mockstream = "0.0.3"
 [features]
 testing = ["parsec-interface/testing", "no-fs-permission-check"]
 no-fs-permission-check = []
+
+[patch.crates-io]
+parsec-interface = { git = "https://github.com/joechrisellis/parsec-interface-rs.git", branch = "add-list-keys" }
```
Comment (hug-dev) [inline Cargo.toml:31]: This should work as everything should be merged in the interface 😃 
```diff
-parsec-interface = { git = "https://github.com/joechrisellis/parsec-interface-rs.git", branch = "add-list-keys" }
+parsec-interface = { git = "https://github.com/parallaxsecond/parsec-interface-rs" }
```

**Review by hug-dev — APPROVED** [review]
Thanks!
I would personally be ok to merge this, even if it contains a `patch`. We can't publish a crates-io version with a `patch` in it anyway (so no mistake is possible) and that leaves us some room to continue development until the next version without keeping this PR up.

@ionut-arm What do you think?

File: Cargo.toml [hunk 2]
```diff
@@ -26,3 +26,6 @@ mockstream = "0.0.3"
 [features]
 testing = ["parsec-interface/testing", "no-fs-permission-check"]
 no-fs-permission-check = []
+
+[patch.crates-io]
+parsec-interface = { git = "https://github.com/parallaxsecond/parsec-interface-rs.git" }
```
Comment (ionut-arm) [inline Cargo.toml:31]: Is it possible to scope this down to a particular commit? Otherwise we could end up making breaking changes.

Comment (joechrisellis) [reply Cargo.toml]: Done.

**Review by ionut-arm — APPROVED** [review]
Thanks!!

**Pull request merged**